webapp/packages/plugin-sql-editor-navigation-tab-script/src/ResourceSqlDataSource.ts

```diff
@@ -92,6 +92,11 @@ export class ResourceSqlDataSource extends BaseSqlDataSource {
     return [ESqlDataSourceFeatures.script, ESqlDataSourceFeatures.query, ESqlDataSourceFeatures.executable, ESqlDataSourceFeatures.setName];
   }
 
+  get hasChanges() {
+    return this._script !== this._previous;
+  }
```

Comment:
`BaseSqlDataSource` have `isSaved` flag, probably you don't need `hasChanges` you can use `isSaved`